woocommerce/wc-admin
# Updating Stripe key field validation to support test keys (#4391)

Fixes #4391

Currently the form to configure Stripe keys accessed via the task list presents an error when test keys are used.

To resolve I've just updated the regex used to validate the Stripe keys to include the pattern(s) used for test keys as well. These are similar in format but have a `pk_test_abc` instead of `pk_live_abc`. I've also added a `[a-zA-Z0-9_]+` to the end of the regex to enforce that at least one additional alpha-numeric character is provided. 

I'd originally also expanded on the validation messages to include more detail into what was a valid key, but I noticed that the UI only really supports single-line validation strings. If they're too long then they overlap the field/button below on a mobile view. I've included a screenshot of this in the section below.

To resolve that I simply shortened the validation messages and am not currently providing much detail to the user on what exactly is wrong. I could plausibly throw in a CSS tweak and expand the validation messages again if that's a preferred solution, although it would likely be a bit hacky since absolute positioning is used for the validation messages.

### Screenshots
**With long validation messages**:
![Screenshot_2020-09-25_15-50-53](https://user-images.githubusercontent.com/444632/94324302-1b5f8c00-ff4e-11ea-8fd3-e91e2b74bd9c.png)

**With shortened validation messages (current state)**:
![image](https://user-images.githubusercontent.com/444632/94324414-a2acff80-ff4e-11ea-8daf-eb41cbc4a3df.png)


### Detailed test instructions:

- Pick a Stripe supported country.
- Make sure you have not selected "CBD" as an industry previously in the profiler.
- Go to the task list payments task.
- Click on Stripe
- Enter "pk_test_abc" in first field. No error should display.
- enter "sk_test_abc" in second field. No error should display.
